src/utils/hooks/useLoadApplication.tsx

```diff
@@ -2,32 +2,58 @@ import { useEffect, useState } from 'react'
 import {
   Application,
   ApplicationStageStatusAll,
+  TemplateStage,
   useGetApplicationQuery,
+  useGetApplicationStatusQuery,
 } from '../../utils/generated/graphql'
 import useTriggerProcessing from '../../utils/hooks/useTriggerProcessing'
 import { getApplicationSections } from '../helpers/application/getSectionsPayload'
-import { ApplicationDetails, TemplateSectionPayload, UseGetApplicationProps } from '../types'
+import {
+  ApplicationDetails,
+  ApplicationStages,
+  TemplateSectionPayload,
+  UseGetApplicationProps,
+} from '../types'
 
-const useLoadApplication = (props: UseGetApplicationProps) => {
-  const { serialNumber } = props
+const useLoadApplication = ({ serialNumber, networkFetch }: UseGetApplicationProps) => {
   const [application, setApplication] = useState<ApplicationDetails>()
   const [templateSections, setSections] = useState<TemplateSectionPayload[]>([])
+  const [appStages, setAppStages] = useState<ApplicationStages>()
   const [isApplicationLoaded, setIsApplicationLoaded] = useState(false)
+  const [checkTrigger, setCheckTrigger] = useState(false)
```

Comment:
So, just to check I'm understanding this correctly:
- we have 3 flags: `isApplicationLoaded`, `checkTrigger` and `triggerProcessing`, which are initially set to `false` (or `true` for `triggerProcessing`)
- after initial application is loaded, `checkTrigger` gets set to `true`
- this allows `useTriggerProcessing` to check the trigger status
- if trigger is ok (not "Processing), then `triggerProcessing` is set to `false`
- this allows `useGetApplicationStatusQuery` to load the status info (fresh from the network) and sets `isApplicationLoaded` to `true`

I think the name `checkTrigger` is a bit confusing and not consistent with the others. I'm not sure what the best solution is here, because `isApplicationLoaded` is kind of ambiguous, since there's two separate stages for loading the application. How about:
- `checkTrigger` becomes `isApplicationLoaded` (or `isApplicationLoadedInitial` -- though probably a bit long)
- `triggerProcessing` stays the same (or `isTriggerProcessing` for full consistency)
- `isApplicationLoaded` becomes `isApplicationReady`

Have a think and see what you reckon. Can have a brief discussion if you want.

Reply:
Finally found time to do that! I used exactly your suggestions, so now there are 3 flags:
* isApplicationLoaded
* isTriggerProcessing
* isApplicationReady

And the flags go in the same order as listed here, each next hook will wait until the previous flag is finished. Then the UI only render on `isApplicationReady`. Also did a similar approach for the review loading hooks